src/main/java/com/arpnetworking/clusteraggregator/aggregation/AggMessageExtractor.java

```diff
@@ -52,6 +56,20 @@ public String entityId(final Object message) {
                     .append(metricData.getMetric())
                     .append("||")
                     .append(metricData.getPeriod());
+            final ArrayList<Messages.DimensionEntry> sortedDimensions = new ArrayList<>(metricData.getDimensionsList());
+            sortedDimensions.sort(Comparator.comparing(Messages.DimensionEntry::getKey));
```

Comment:
So comparing only on the key assumes that each key only exists once (otherwise multiple values for that key are in arbitrary order). So if we're going to make this assumption we need to check for duplicates (e.g. add them to a TreeMap). 

@BrandonArp why did we not use an actual map in the protocol definition? Are we planning to support multimaps for dimensions? 

Doc:
https://developers.google.com/protocol-buffers/docs/proto#maps

Protocol:
https://github.com/ArpNetworking/metrics-aggregator-protocol/blob/master/src/main/proto/messages.proto#L54